apps/sms/services/js/conversation/conversation_client.js

```diff
@@ -32,36 +27,35 @@
    */
   var ConversationClient = {
     /**
-     * Reference to active bridge client instance.
-     * @type {Client}
-     */
-    [priv.client]: null,
-
-    /**
-     * Unique identifier of app instance where client resides in.
-     * @type {string}
-     */
-    [priv.appInstanceId]: null,
-
-    /**
      * Initialized conversation service client bridge.
      * @param {string} appInstanceId Unique identifier of app instance where
      * client resides in.
      */
     init(appInstanceId) {
-      if (!appInstanceId) {
-        throw new Error('AppInstanceId is required!');
-      }
+      // We use window based ConversationService initially, so let's init it.
+      exports.ConversationService.init();
```

Comment:
That's what I'm not sure of, but one of the ideas is to keep all this switching, window-context-hack knowledge inside client only... as well path to the SharedWorker's main file.